Magento2/Sniffs/PHP/AutogeneratedClassNotInConstructorSniff.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types = 1);
+
+namespace Magento2\Sniffs\PHP;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+/**
+ * Detects possible usage of 'var' language construction.
+ */
+class AutogeneratedClassNotInConstructorSniff implements Sniff
+{
+    private const ERROR_CODE = 'AUTOGENERATED_CLASS_NOT_IN_CONSTRUCTOR';
+
+    /**
+     * @var array
+     */
+    private $constructorParameters = [];
+
+    /**
+     * @var array
+     */
+    private $uses = [];
+
+    /**
+     * @inheritdoc
+     */
+    public function register()
+    {
+        return [T_FUNCTION, T_DOUBLE_COLON, T_USE];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        if ($phpcsFile->getTokens()[$stackPtr]['type'] === 'T_USE') {
+            $this->registerUses($phpcsFile, $stackPtr);
+        }
+        if ($phpcsFile->getTokens()[$stackPtr]['type'] === 'T_FUNCTION') {
+            $this->registerConstructorParameters($phpcsFile, $stackPtr);
+        }
+        if ($phpcsFile->getTokens()[$stackPtr]['type'] === 'T_DOUBLE_COLON') {
+            if (!$this->isObjectManagerGetInstance($phpcsFile, $stackPtr)) {
+                return;
+            }
+
+            $statementStart = $phpcsFile->findStartOfStatement($stackPtr);
+            $statementEnd = $phpcsFile->findEndOfStatement($stackPtr);
+            $equalsPtr = $phpcsFile->findNext(T_EQUAL, $statementStart, $statementEnd);
+
+            if (!$equalsPtr) {
+                return;
+            }
+
+            $variableInParameters = false;
+            if ($variable = $phpcsFile->findNext(T_VARIABLE, $equalsPtr, $statementEnd)) {
```

Comment:
This method is too long, please consider decomposition into several private functions